### PR TITLE
{Service Connector} `az webapp connection create sql`: auto install extension `serviceconnector-passwordless`

### PR DESCRIPTION
**Pull request by xfz11**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 1]
```diff
@@ -293,8 +299,39 @@ def connection_create(cmd, client,  # pylint: disable=too-many-locals,too-many-s
...
+    if secret_auth_info_auto is not None:
+        all_auth_info.append(secret_auth_info_auto)
+    if user_identity_auth_info is not None:
+        all_auth_info.append(user_identity_auth_info)
+    if system_identity_auth_info is not None:
+        all_auth_info.append(system_identity_auth_info)
+    if service_principal_auth_info_secret is not None:
+        all_auth_info.append(service_principal_auth_info_secret)
+    if not new_addon and len(all_auth_info) != 1:
+        raise ValidationError('Only one auth info is needed')
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:314]: duplicate logics also in `local_connection_create`, `connection_create_func`, and `local_connection_create_func`consider doing some refractoring to reuse

**Review by houk-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 1 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: fixed

**Review by houk-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 import time
 from knack.log import get_logger
-from knack.util import todict
+from knack.util import todict, CLIError
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py:8]: CLIError is deprecated.

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py [hunk 3]
```diff
@@ -347,3 +347,91 @@ def get_object_id_of_current_user():
...
+    except ExtensionNotInstalledException:
+        return _install_extension(cmd, extension_name)
+    return True
+
+
+def _update_extension(cmd, extension_name):
+    from azure.cli.core.extension import ExtensionNotInstalledException
+    try:
+        from azure.cli.core.extension import operations
+        operations.update_extension(cmd=cmd, extension_name=extension_name)
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py:416]: @kairu-ms This core function throw `CLIError `and we just catch the error

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py (hunk 2 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py:8]: We don't throw CLIError but just catch the existing one.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**